lib/memory.gi

```diff
@@ -228,8 +228,10 @@ InstallMethod( \^, "objects with memory", true,
   function(a,b)
     local slp;
     slp := a!.slp;
-    if a!.n = 0 then
+    if a!.n = 0 or b = 0 then
       return ObjWithMemory( slp, 0, a!.el );
+    elif b = 1 then
+      return ObjWithMemory( slp, a!.n, a!.el );
```

Comment:
Can't we simply return `a`?
The entries in the result of `GeneratorsWithMemory` are immutable. Do we have situations where objects in `IsObjWithMemory` are mutable?
(As far as I see, all this is not really documented.)